addons/event_sale/wizard/event_configurator.py

```diff
@@ -11,3 +12,33 @@ class EventConfigurator(models.TransientModel):
     product_id = fields.Many2one('product.product', string="Product", readonly=True)
     event_id = fields.Many2one('event.event', string="Event")
     event_ticket_id = fields.Many2one('event.event.ticket', string="Event Ticket")
+    has_available_events = fields.Boolean(
+        compute="_compute_has_available_events",
+        help="Technical field used to know if the user has at least one event that can be selected and added to the sales order")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(EventConfigurator, self).default_get(fields)
+        if 'event_id' in fields:
+            domain = [('event_ticket_ids.product_id', '=', res['product_id']),
+                      ('date_end', '>=', date.today().strftime('%Y-%m-%d 00:00:00')),
+                      ('stage_id.pipe_end', '=', False), ]
+            event = self.env['event.event'].search(domain)
```

Comment:
I think you can inline the domain.

Side note, you could set a limit=2 here to avoid fetching a lot of records unnecessarily.
I don't think it will save much processing time since there won't be that many events but every bit counts.

Also rename "event" to "upcoming_events" ?

Reply:
done, the method has been changed.